### DISCUSSION TITLE
Global State implementation

### DISCUSSION THREAD

**Discussion by IrisHerholz** [neuefische/questions#479] [Web]
Hello, everyone!
I would like the entry onthe "new entry page" to appear on the "entrylist page" after it has been submitted. 
I was advised to implement Global States for this, but I can't get any results.
Attached is my current status, I commented out everything that concerns global state so that the rest works.
Thanks for any advice!
https://github.com/IrisHerholz/capstone-project/pull/11

**Comment by ThomasF85** [comment]
Hi Iris,

for your global state with jotai to work you need a couple of things:

1.: You have to create an atom that you can import wherever you need the state:
```
import { atom } from "jotai";
export const entries = atom([]);
```

2.: You have to use useAtom where you want to actually update / add an entry to your journal inside a component (make sure useAtom is only used inside a component!)
```
import {entries} from "somewhere/in/my/project";
import { useAtom } from "jotai";

export default function MyComponent() {
  const [journalEntries, setJournalEntries] = useAtom(entries);

  function addEntry(newEntry) {
    setJournalEntries(previousEntries => [...previousEntries, newEntry]);
  }

   return ...
}
```

3.: You have to use useAtom where you want to show the journal entries (make sure useAtom is only used inside a component!)
```
import {entries} from "somewhere/in/my/project";
import { useAtom } from "jotai";

export default function MyListComponent() {
  const [journalEntries, setJournalEntries] = useAtom(entries);

   return (
      <ul>
        {journalEntries.map(entry => (
           <li key={entry.id}>
              ...
           </li>
        ))}
      </ul>
   )
}
```

If you update / add your entries on one page then you will see the updated list on every page where they are used.

**Comment by IrisHerholz** [comment]
![Bildschirmfoto 2023-02-01 um 16 23 26](https://user-images.githubusercontent.com/118169582/216084918-dacabf90-36eb-4e89-8ba2-200776a758cf.png)


Thanks for the answer.
At what point must the error be? I implemented everything as discussed yesterday. Unfortunately I can't get it solved and there are no further error messages

**Reply by ThomasF85** [reply]
Please provide the whole code of your component or a Link to a current PR with the code, if it exists.

**Reply by IrisHerholz** [reply]
https://github.com/IrisHerholz/capstone-project/pull/11

**Reply by ThomasF85** [reply]
You are using a wrong import in your Form component:
```
import { useAtom } from "react";
```

**Comment by IrisHerholz** [comment]
[](https://github.com/IrisHerholz/capstone-project/pull/11)
